### PR TITLE
Add `cancel` and `flush` support to debounced functions

### PR DESCRIPTION
**Pull request by nikku**
Related to https://github.com/bpmn-io/form-js/pull/157

### CODE REVIEW

File: lib/fn.js [hunk 1]
```diff
@@ -9,34 +12,49 @@
...
 
+    timer = lastNow = lastArgs = lastThis = undefined;
+  }
+
+  function flush() {
+    if (timer) {
+      fire(true);
+    }
+
+    clear();
```
Comment (barmac) [inline lib/fn.js:54]: We are calling `clear()` [twice if timer is set](https://github.com/bpmn-io/min-dash/pull/20/files#diff-1b3f81b82a473a44ed3c5f5e16fce0030abc3b0cdf5efac082ba1fc01a9de490R34)

Comment (nikku) [reply lib/fn.js:54]: I could also go for this one here: 

```
  function flush() {
    if (timer) {
      fire(true);
    } else {
      clear();
    }
  }
```

But would that not make this piece harder to reason about?

Comment (nikku) [reply lib/fn.js:54]: Obviously we could call the method `fireAndClear` but then... it is only internals.

**Review by barmac — APPROVED** [review]

File: lib/fn.js (hunk 1 above)
Comment (barmac) [reply lib/fn.js:54]: ok let's leave it as it is

**Pull request merged**